### ISSUE TITLE
pdns-recursor-4.3.0 build broken on illumos

### ISSUE THREAD

**Issue by jperkin** [PowerDNS/pdns#9025]
- Program: Recursor
 - Issue type: Bug report

### Short description

```
portsmplexer.cc: In function 'FDMultiplexer* makePorts()':
portsmplexer.cc:43:33: error: invalid new-expression of abstract class type 'PortsFDMultiplexer'
   return new PortsFDMultiplexer();
                                 ^
portsmplexer.cc:17:7: note:   because the following virtual functions are pure within 'PortsFDMultiplexer':
 class PortsFDMultiplexer : public FDMultiplexer
       ^~~~~~~~~~~~~~~~~~
In file included from portsmplexer.cc:9:0:
mplexer.hh:84:16: note: 	virtual void FDMultiplexer::getAvailableFDs(std::vector<int>&, int)
   virtual void getAvailableFDs(std::vector<int>& fds, int timeout) = 0;
                ^~~~~~~~~~~~~~~
In file included from portsmplexer.cc:9:0:
mplexer.hh:163:23: note: 	virtual std::__cxx11::string FDMultiplexer::getName() const
   virtual std::string getName() const = 0;
                       ^~~~~~~
gmake[2]: *** [Makefile:1522: portsmplexer.o] Error 1
```

### Environment

 - Operating system: SmartOS/illumos, and probably Solaris
 - Software version: 4.3.0
 - Software source: pkgsrc

### Steps to reproduce

```
cd pkgsrc/net/powerdns-recursor
bmake
```
### Expected behaviour

Build works.

### Actual behaviour

Build error log available here: https://us-east.manta.joyent.com/pkgsrc/public/reports/upstream-trunk/20200413.2254/powerdns-recursor-4.3.0/build.log

### Other information

Sorry, I don't have time to dig into this at the moment, so have just resorted to creating this bug report for now.  Hacking the configure script to disable `have_solaris` makes the build succeed, presumably using a poll backend or similar instead of event ports.

**Comment by Habbie** [comment]
Is it easy for you to kick off a build with a patch, if we give you one?

**Comment by jperkin** [comment]
Yeh, trivial.

**Comment by Habbie** [comment]
Hi Jonathan, @rgacogne fixed this in #9031, and we'll backport that for the next minor, I'm sure. Perhaps you want to take the patch into pkgsrc for now?

**Comment by jperkin** [comment]
Yep looks good, thanks! I've pulled this into pkgsrc in <https://github.com/NetBSD/pkgsrc/commit/76dafceb03e0a47b2e4f7798cc67b88abaf68dc0>.
